### ISSUE TITLE
Infinite loop via modifying loop index

### ISSUE THREAD

**Issue by smarx** [ethereum/vyper#1708]
The generated variable `_index_for_*` shouldn't be accessible. The following generates an infinite loop:

```python
@public
def infinite_loop():
    for n in [1,2,3]:
        _index_for_n = 0 # BUG
```

**Comment by fubuloubu** [comment]
Might be fixed by #1601

**Comment by charles-cooper** [comment]
confirmed fixed in master (4666a872acae10efed0f9c349e9359e686bd3a08)
```
$ vyper -f ir /dev/stdin <<EOF
@public
def infinite_loop():
    for n in [1,2,3]:
        _index_for_n = 0 # BUG
EOF

Error compiling: /dev/stdin
vyper.exceptions.VariableDeclarationException: line 4:8 Variable type not defined
     3     for n in [1,2,3]:
---> 4         _index_for_n = 0 # BUG
---------------^
```
